lib/samples/apply_symbology_to_shapefile/apply_symbology_to_shapefile.dart

```diff
@@ -0,0 +1,146 @@
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'dart:io';
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/common/common.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/utils/sample_data.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/utils/sample_state_support.dart';
+import 'package:flutter/material.dart';
+import 'package:path_provider/path_provider.dart';
+
+class ApplySymbologyToShapefile extends StatefulWidget {
+  const ApplySymbologyToShapefile({super.key});
+
+  @override
+  State<ApplySymbologyToShapefile> createState() =>
+      _ApplySymbologyToShapefileState();
+}
+
+class _ApplySymbologyToShapefileState extends State<ApplySymbologyToShapefile>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // Hold reference to the feature layer so that its renderer can be changed when button is pushed.
+  late FeatureLayer _shapefileFeatureLayer;
+  // Hold reference to default renderer to enable switching back.
+  Renderer? _defaultRenderer;
+  // Hold reference to alternate renderer to enable switching.
+  SimpleRenderer? _alternateRenderer;
+  // Create variable for holding state relating to the renderer.
+  var _alternate = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    const Text('Alternate renderer'),
+                    Switch(
+                      value: _alternate,
+                      onChanged: updateRenderer,
+                    ),
+                  ],
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
```

Comment:
There's a new lint rule that will want the return value to be `Future<void>` instead of just `void`.